### PR TITLE
Bugfix FXIOS-9174 [Multi-window] Fix theme bugs

### PR DESCRIPTION
**Pull request by mattreaganmozilla**
## :scroll: Tickets
[Jira ticket](https://mozilla-hub.atlassian.net/browse/FXIOS-9174)
[Github issue](https://github.com/mozilla-mobile/firefox-ios/issues/20332)

## :bulb: Description

This fixes several bugs with theme changes across multiple iPad windows.

## :pencil: Checklist
You have to check all boxes before merging
- [x] Filled in the above information (tickets numbers and description of your work)
- [x] Updated the PR name to follow our [PR naming guidelines](https://github.com/mozilla-mobile/firefox-ios/wiki/Pull-Request-Naming-Guide)
- [ ] Wrote unit tests and/or ensured the tests suite is passing
- [ ] When working on UI, I checked and implemented accessibility (minimum Dynamic Text and VoiceOver)
- [ ] If needed, I updated documentation / comments for complex code and public methods
- [ ] If needed, added a backport comment (example `@Mergifyio backport release/v120`)

### CODE REVIEW

**Comment by mattreaganmozilla** [comment]
@OrlaM @adudenamedruby Still testing a few things, so adding `Do Not Merge` temporarily, but there aren't any remaining bugs that I can see yet so this should be ready for review.

File: firefox-ios/Client/Extensions/UIView+ThemeUUIDIdentifiable.swift [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+import Shared
+
+extension UIView: ThemeUUIDIdentifiable {
+    /// Convenience that allows most UIViews to automatically reference their parent window UUID.
+    /// If the view is installed in a view hierarchy that is part of a window, or is a type
+    /// that has its UUID injected, this returns the window UUID. If not, returns nil.
+    public var currentWindowUUID: WindowUUID? {
+        if let injectedUUID = (self as? InjectedThemeUUIDIdentifiable)?.windowUUID { return injectedUUID }
+        if let tableView = (self as? UITableView),
+           let uuidProvider = (tableView.delegate as? InjectedThemeUUIDIdentifiable) { return uuidProvider.windowUUID }
```
Comment (adudenamedruby) [inline firefox-ios/Client/Extensions/UIView+ThemeUUIDIdentifiable.swift:16]: nit for multi-line if statements

```suggestion
        if let tableView = (self as? UITableView),
           let uuidProvider = (tableView.delegate as? InjectedThemeUUIDIdentifiable) {
            return uuidProvider.windowUUID
        }
```

**Review by adudenamedruby — APPROVED** [review]
LGTM

File: firefox-ios/Client/Extensions/UIView+ThemeUUIDIdentifiable.swift (hunk 1 above)
Comment (mattreaganmozilla) [reply firefox-ios/Client/Extensions/UIView+ThemeUUIDIdentifiable.swift]: Fixed; these should be cleaned up now, thanks @adudenamedruby .

**Pull request merged**